ibis/backends/bigquery/datatypes.py

```diff
@@ -1,101 +1,124 @@
 from __future__ import annotations
 
 import google.cloud.bigquery as bq
-import sqlglot as sg
+import sqlglot.expressions as sge
 
 import ibis
 import ibis.expr.datatypes as dt
 import ibis.expr.schema as sch
-from ibis.formats import SchemaMapper, TypeMapper
-
-_from_bigquery_types = {
-    "INT64": dt.Int64,
-    "INTEGER": dt.Int64,
-    "FLOAT": dt.Float64,
-    "FLOAT64": dt.Float64,
-    "BOOL": dt.Boolean,
-    "BOOLEAN": dt.Boolean,
-    "STRING": dt.String,
-    "DATE": dt.Date,
-    "TIME": dt.Time,
-    "BYTES": dt.Binary,
-    "JSON": dt.JSON,
-}
-
-
-class BigQueryType(TypeMapper):
-    @classmethod
-    def to_ibis(cls, typ: str, nullable: bool = True) -> dt.DataType:
-        if typ == "DATETIME":
-            return dt.Timestamp(timezone=None, nullable=nullable)
-        elif typ == "TIMESTAMP":
-            return dt.Timestamp(timezone="UTC", nullable=nullable)
-        elif typ == "NUMERIC":
-            return dt.Decimal(38, 9, nullable=nullable)
-        elif typ == "BIGNUMERIC":
-            return dt.Decimal(76, 38, nullable=nullable)
-        elif typ == "GEOGRAPHY":
-            return dt.GeoSpatial(geotype="geography", srid=4326, nullable=nullable)
-        else:
-            try:
-                return _from_bigquery_types[typ](nullable=nullable)
-            except KeyError:
-                raise TypeError(f"Unable to convert BigQuery type to ibis: {typ}")
+from ibis.backends.base.sqlglot.datatypes import SqlglotType
+from ibis.formats import SchemaMapper
+
+
+class BigQueryType(SqlglotType):
+    dialect = "bigquery"
+
+    default_decimal_precision = 38
+    default_decimal_scale = 9
+
+    @classmethod
+    def _from_sqlglot_NUMERIC(cls) -> dt.Decimal:
+        return dt.Decimal(
+            cls.default_decimal_precision,
+            cls.default_decimal_scale,
+            nullable=cls.default_nullable,
+        )
+
+    @classmethod
+    def _from_sqlglot_BIGNUMERIC(cls) -> dt.Decimal:
+        return dt.Decimal(76, 38, nullable=cls.default_nullable)
+
+    @classmethod
+    def _from_sqlglot_DATETIME(cls) -> dt.Decimal:
+        return dt.Timestamp(timezone=None, nullable=cls.default_nullable)
+
+    @classmethod
+    def _from_sqlglot_TIMESTAMP(cls) -> dt.Decimal:
+        return dt.Timestamp(timezone="UTC", nullable=cls.default_nullable)
+
+    @classmethod
+    def _from_sqlglot_GEOGRAPHY(cls) -> dt.Decimal:
+        return dt.GeoSpatial(
+            geotype="geography", srid=4326, nullable=cls.default_nullable
+        )
+
+    @classmethod
+    def _from_sqlglot_TINYINT(cls) -> dt.Int64:
+        return dt.Int64(nullable=cls.default_nullable)
+
+    _from_sqlglot_UINT = (
```

Comment:
This is hard to read, I assume ruff formats it this way. 
Perhaps we should have a way to update the type mapping in a more convenient way.